File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd2.m
%Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use.
function testSendQuadCmd2()

%open the interface
KQUAD.dev    = '/dev/ttyUSB0';
KQUAD.driver = @kQuadInterfaceAPI;
KQUAD.baud   = 1000000;
KQUAD.id     = 1;
KQUAD.chan   = 1;
KQUAD.type   = 0; %0 for standard, 1 for nano

KQUAD.driver('connect',KQUAD.dev,KQUAD.baud);


while(1)
  rpm = [3000 3000 3000 3000]; %four rpm for the motors
  
  KQUAD.driver('SendQuadCmd2',KQUAD.id, KQUAD.chan, KQUAD.type, rpm);

  pause(0.03);
end

File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd2.cc
/* Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use. */
#include "kQuadInterface.hh"

int main()
{
  char * dev = (char*)"/dev/ttyUSB0";
  int baud   = 1000000;
  
  kQuadInterface kqi;
  
  if (kqi.Connect(dev,baud))
  {
    printf("could not connect to the device\n");
    return -1;
  }
  
  if (kqi.StartSendThread())
  {
    printf("could not start thread\n");
    return -1;
  }
  
  uint8_t quadId   = 1;
  uint8_t quadType = 0; //standard
  uint8_t channel  = 1;
  
  uint16_t rpm0 = 3000;
  uint16_t rpm1 = 3000;
  uint16_t rpm2 = 3000;
  uint16_t rpm3 = 3000;
  
  while(1)
  {
    kqi.SendQuadCmd2(quadId, quadType, channel, rpm0, rpm1, rpm2, rpm3);
    
    usleep(50000);
    
    printf("."); fflush(stdout);
  }
  
 
  return 0;
}
